app/Dockerfile

```diff
@@ -1,10 +1,8 @@
 FROM eclipse-temurin:17-jre-alpine
 
 # hadolint ignore=DL3018
-RUN apk update && \
-    apk --no-cache add curl redis sudo gcompat && \
-    apk upgrade --available && \
-    rm -rf /var/cache/apk/*
```

Comment:
Why is the rm -rf line removed?

Reply:
Here is why I made those changes: 

1. I do see your concern of removing `apk update` since this could potentially lead to issues as it's responsible for updating the package index. However, when using `apk add --no-cache`, the necessity for a separate apk update command is negated since --no-cache fetches the index during the install process without storing it locally. 
2. In regards to not specifying the package version, you are right since we don't want to hard code versions to our docker image. So I will change this to: `libxml2`
3. Finally, though removing the package manager cache with this command line `rm -rf /var/cache/apk/*` is best practice for reducing Docker images sizes, the reason for removing this line is again back to the `apk add --no-cache` command since the cache is not preserved  and manually removing the cache directory becomes redundant at this point.

Reply:
I stand corrected!  Thanks for the explaination Ponnia, I didn't realize that the "--no-cache" flag was such a direct and efficient replacement for the "update and remove" pattern that is so prevalent.  It makes me wonder how many code samples on the internet make the same mistake i did by missing the true meaning of "--no-cache". Thanks again!